recipes/transformer-engine/meta.yaml

```diff
@@ -0,0 +1,99 @@
+{% set version = "1.13" %}
+{% set torch_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
+
+{% if cuda_compiler_version in (None, "None", True, False) %}
+{% set cuda_major = 0 %}
+{% else %}
+{% set cuda_major = environ.get("cuda_compiler_version", "11.8").split(".")[0] | int %}
+{% endif %}
+
+package:
+  name: transformer_engine
+  version: {{ version }}
+
+source:
+  - url: https://github.com/NVIDIA/TransformerEngine/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: 90dd8f694cba29707c502a4f4d12871cb9d4c1f157d329f22cb3d382de8468ae
+
+  - url: https://github.com/NVIDIA/cudnn-frontend/archive/refs/tags/v1.8.0.tar.gz
+    sha256: 5907bf8baa8f3b74ad430d54027aadd14a0a61166ae4b00c2daece0df384409e
+    folder: 3rdparty/cudnn-frontend
```

Comment:
Is it ok to vendor `cudnn-frontend` here? I've packaged this at https://github.com/conda-forge/python-cudnn-frontend-feedstock and have been trying to get it to work (unsuccessfully) at https://github.com/conda-forge/staged-recipes/pull/26296#issuecomment-2113409609. But this solution looks much simpler :slightly_smiling_face:

Reply:
Yeah that's a good question. In conversations with the TE team, they work pretty closely with cudnn-frontend and sometimes have odd tags or versions here. I also don't think the repo is set up to be able to pull these dependencies from a source outside `3rdparty`. That might be something we could try to fix in subsequent versions though. `cudnn-frontend` is a header-only library though, so it should be fairly lightweight.

Reply:
From https://conda-forge.org/docs/maintainer/guidelines/#generalities

> The source should not contain vendored code. If it does:
>    Package the vendored project separately and specify the needed dependency in the requirements section. Preferred if the vendored code is needed at runtime.
>    Allow the vendored code, but make sure the license files are included in the about.license field. Usually ok if it's only a build-time dependency (e.g. headers-only library)
```

Reply:
Revending headers for compile-time is OK. What's not OK is revending them at runtime. This package is shipping a copy of CUDNN in $PREFIX/targets.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1235681&view=logs&j=df9408f7-e795-500f-143d-aa7735a0ed8b&t=47addc80-ccbf-566b-ee3a-66cdb8436937&l=4381

Reply:
Ah, yes, that is a problem. I moved some headers around so the CMAKE command could find them, but didn't realize they were getting re-distributed. Since we're patching those cmake build flags anyways I'll try to point cmake to the right place to find them in these conda builds

Reply:
Well it's not easy to get the cmake command to find these in a different location. The alternative could just be to clean these up post-build, right? Let me try doing that.